aiohttp/http_websocket.py

```diff
@@ -626,28 +627,49 @@ async def _send_frame(
         if (compress or self.compress) and opcode < 8:
             if compress:
                 # Do not set self._compress if compressing is for this frame
-                compressobj = ZLibCompressor(
-                    level=zlib.Z_BEST_SPEED,
-                    wbits=-compress,
-                    max_sync_chunk_size=WEBSOCKET_MAX_SYNC_CHUNK_SIZE,
-                )
+                compressobj = self._make_compress_obj(compress)
+                message = await compressobj.compress(message)
+                self._write_compressed_message(message, opcode, compressobj, rsv)
             else:  # self.compress
+                # If we are reusing the compress object, we need to hold a lock
+                # around the compress/flush/write to ensure that messages get
+                # compressed in order and takeover is not violated.
                 if not self._compressobj:
-                    self._compressobj = ZLibCompressor(
-                        level=zlib.Z_BEST_SPEED,
-                        wbits=-self.compress,
-                        max_sync_chunk_size=WEBSOCKET_MAX_SYNC_CHUNK_SIZE,
-                    )
+                    self._compressobj = self._make_compress_obj(self.compress)
                 compressobj = self._compressobj
+                if not self._compress_lock:
+                    self._compress_lock = asyncio.Lock()
```

Comment:
Shall we just do this in the above if block along with `self._compressobj`, save us an extra check.

Reply:
Good idea, done